### PR TITLE
Http extract step

### PR DESCRIPTION
**Pull request by cameronmarsh**

### CODE REVIEW

**Review by LtChae — APPROVED** [review]
Looks workable, though the amount of code needed to implement just the http step in schemas is worrying to me. We need to be able to make new steps fairly easily if this is how we add new ingestion functionality going forward, to say nothing of converting the steps we already have.

File: apps/andi/lib/andi/input_schemas/datasets.ex [hunk 1]
```diff
@@ -111,7 +113,7 @@ defmodule Andi.InputSchemas.Datasets do
       |> StructTools.to_map()
       |> Map.merge(form_changes)
 
-    new_changes = %{technical: technical_changes, business: business_changes, id: dataset_id} |> StructTools.to_map()
+    new_changes = %{technical: technical_changes, business: business_changes, id: dataset_id}
```
Comment (jessie-morris) [inline apps/andi/lib/andi/input_schemas/datasets.ex:116]: Why no to_map anymore?

**Comment by cameronmarsh** [comment]
In response to Tim's comment, unfortunately a lot of this is required to make new Ecto tables and liveview components, so there's not a lot we can do there. However, some of it is only temporary to allow both the sourceUrl and extract step forms to be shown at the same time. For future extract step cards at least, there are patterns that can be extracted out and shared (like the form_section file) that will result in less code.

File: apps/andi/lib/andi/input_schemas/datasets.ex (hunk 1 above)
Comment (cameronmarsh) [reply apps/andi/lib/andi/input_schemas/datasets.ex:116]: They are all maps already, this was doing nothing

**Pull request merged**